base_layer/core/src/transactions/transaction_components/encrypted_data.rs

```diff
@@ -99,6 +100,10 @@ impl EncryptedData {
         let aead_key = kdf_aead(encryption_key, commitment);
         let cipher = XChaCha20Poly1305::new(GenericArray::from_slice(aead_key.reveal()));
         let mut ciphertext = cipher.encrypt(nonce_ga, aead_payload)?;
+
+        // Openings contains the mask so we need to zeroize it
+        openings.zeroize();
```

Comment:
Is this sufficient, since the previous line can fail without zeroizing the opening?

Reply:
Well spotted!